commit 7c4bf8485ad3cd17dbc96402229f1b51e123fc64
Author: Tim Gover <tim.gover@raspberrypi.org>
Date:   Fri Jun 22 14:08:55 2018 +0100

    ASoC: rpi-dac: Move snd_soc_dai_set_bclk_ratio to .init
    
    snd_soc_dai_set_bclk_ratio is called with a fixed value of 64. This can
    be called from .init instead allowing the more verbose hw_params
    callback method to be removed.
    
    Signed-off-by: Tim Gover <tim.gover@raspberrypi.org>

diff --git a/sound/soc/bcm/rpi-dac.c b/sound/soc/bcm/rpi-dac.c
index 900889629661..978ea1155f00 100644
--- a/sound/soc/bcm/rpi-dac.c
+++ b/sound/soc/bcm/rpi-dac.c
@@ -25,23 +25,9 @@
 
 static int snd_rpi_rpi_dac_init(struct snd_soc_pcm_runtime *rtd)
 {
-	return 0;
+	return snd_soc_dai_set_bclk_ratio(rtd->cpu_dai, 32*2);
 }
 
-static int snd_rpi_rpi_dac_hw_params(struct snd_pcm_substream *substream,
-				       struct snd_pcm_hw_params *params)
-{
-	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct snd_soc_dai *cpu_dai = rtd->cpu_dai;
-
-	return snd_soc_dai_set_bclk_ratio(cpu_dai, 32*2);
-}
-
-/* machine stream operations */
-static struct snd_soc_ops snd_rpi_rpi_dac_ops = {
-	.hw_params = snd_rpi_rpi_dac_hw_params,
-};
-
 static struct snd_soc_dai_link snd_rpi_rpi_dac_dai[] = {
 {
 	.name		= "RPi-DAC",
@@ -52,7 +38,6 @@ static struct snd_soc_dai_link snd_rpi_rpi_dac_dai[] = {
 	.codec_name	= "pcm1794a-codec",
 	.dai_fmt	= SND_SOC_DAIFMT_I2S | SND_SOC_DAIFMT_NB_NF |
 				SND_SOC_DAIFMT_CBS_CFS,
-	.ops		= &snd_rpi_rpi_dac_ops,
 	.init		= snd_rpi_rpi_dac_init,
 },
 };
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1187_7c4bf8485ad3cd17dbc96402229f1b51e123fc64.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1187_7c4bf8485ad3cd17dbc96402229f1b51e123fc64.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
